### ISSUE TITLE
FormEntryActivity needs refactoring.

### ISSUE THREAD

**Issue by heyjamesknight** [getodk/collect#1616]
Currently, FormEntryActivity is very hard to follow, at almost 3K lines. We should start trying to break distinct functionality out into testable classes to reduce this and make it a little easier to think about.

**Comment by akshay-ap** [comment]
@jknightco can you mention the parts of `FormEntryAcitivity` that needs to be split into separate classes and need to add testing?
I can work on it.

**Comment by seadowg** [comment]
So true! We're slowly trying to move code out and look at replacing things like the custom swipe code with a Fragment based ViewPager (there are some notes on that [here](https://gist.github.com/seadowg/93f33419460bcc7895990dab5c5958c5)). 

Closing this for the moment as this work will be incremental and ongoing so we won't be able to say "we're done" at any point.
